backend/src/LibExecution/AnalysisTypes.fs

```diff
@@ -26,20 +22,94 @@ type HashVersion = int
 type FnName = string
 type FunctionResult = FnName * id * FunctionArgHash * HashVersion * RT.Dval
 
-type TraceID = System.Guid
+/// TraceIDs are a UUID with a very specific structure, that is designed so that when
+/// it is lexicographically sorted, that the sort order is reverse chronological.
+/// This allows us to sort traces by time in the Google Cloud Storage, which only
+/// supports lexicographical sorting. It also allows us to omit timestamps in the DB
+/// and use trace ordering and indexing. Traces are also stored in the index better
+/// (as in, more compactly) as a result.
```

Comment:
> Traces are also stored in the index better (as in, more compactly) as a result.

Could you please explain this? (or is that referencing just the omission of the timestamp in the db?)

Reply:
A reason people like ULIDs is that ULIDs are close to each other, and so can be stored more densely in a DB index, vs random UUIDs that are spread across the index.